fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -695,7 +697,13 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         startupPathProvider.onIntentReceived(intent)
     }
 
-    open fun handleNewIntent(intent: Intent) {
+    @VisibleForTesting
+    @Suppress("UndocumentedPublicFunction") // only visible for tests.
```

Comment:
```suggestion
    @Suppress("UndocumentedPublicFunction") // only visible for tests.
```
We should avoid this suppressing UndocumentedPublicFunction. Could we change the visibility of the this function to `internal` instead?

Reply:
As discussed offline, adding kDoc and removing the suppression. Follow-up [task](https://bugzilla.mozilla.org/show_bug.cgi?id=1873009) created to investigate if we want should adjust detekt rules and make this not fail for VisibleForTesting